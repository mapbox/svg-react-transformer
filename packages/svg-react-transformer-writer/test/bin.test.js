'use strict';

const exec = require('child_process').exec;
const del = require('del');
const path = require('path');

const outputDir = path.join(__dirname, '../../../tmp/');
const runWriter = (args, callback) => {
  exec(
    `bin/svg-react-transformer.js ${args.join(' ')}`,
    { cwd: path.join(__dirname, '..') },
    callback
  );
};

describe('bin', () => {
  afterEach(() => {
    return del(outputDir, { force: true });
  });

  test('logs verbose output', done => {
    runWriter(
      ['test/fixtures/*.svg', '--out-dir', outputDir, '--verbose'],
      (error, stdout) => {
        expect(error).toBeFalsy();
        expect(stdout).toMatchSnapshot();
        done();
      }
    );
  });
});
