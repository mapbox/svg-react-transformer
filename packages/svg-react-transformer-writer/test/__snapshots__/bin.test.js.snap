// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin logs verbose output 1`] = `
"test/fixtures/airplane.svg -> ../../tmp/airplane.js
test/fixtures/apple.svg -> ../../tmp/apple.js
test/fixtures/big.svg -> ../../tmp/big.js
Transformed 3 files from SVGs to React component modules
"
`;
